src/main/java/examples/ExampleIngestFunction2.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+package examples;
+
+import com.salesforce.functions.jvm.sdk.Context;
+import com.salesforce.functions.jvm.sdk.InvocationEvent;
+import com.salesforce.functions.jvm.sdk.SalesforceFunction;
+import com.salesforce.functions.jvm.sdk.data.DataApi;
+import com.salesforce.functions.jvm.sdk.data.Record;
+import com.salesforce.functions.jvm.sdk.data.bulk.BulkApi;
+import com.salesforce.functions.jvm.sdk.data.bulk.DataTable;
+import com.salesforce.functions.jvm.sdk.data.bulk.IngestJobReference;
+import com.salesforce.functions.jvm.sdk.data.bulk.IngestOperation;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ExampleIngestFunction2 implements SalesforceFunction<String, List<String>> {
+  @Override
+  public List<String> apply(InvocationEvent<String> event, Context context) throws Exception {
+    DataApi dataApi = context.getOrg().get().getDataApi();
```

Comment:
one thing that was suggested when we did the original work was the construction of `Record` object should be moved out of `dataApi`.  e.g.; `context.getOrg().newRecordBuilder()...` 

are you still considering this or would you rather avoid this breaking change?